packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart

```diff
@@ -239,16 +239,49 @@ class FuchsiaRemoteConnection {
     _pollDartVms = false;
   }
 
+  /// Helper method for [getMainIsolatesByPattern].
+  ///
+  /// Called when either there are no Isolates that exist that match
+  /// `pattern`, or there are not yet any active Dart VM's on the system
+  /// (possible when the Isolate we're attempting to connect to is in the only
+  /// instance of the Dart VM and its service port has not yet opened).
+  Future<List<IsolateRef>> _waitForMainIsolatesByPattern([
+    Pattern pattern,
+    Duration timeout = _kIsolateFindTimeout,
+  ]) async {
+    final Completer<List<IsolateRef>> completer =
+        new Completer<List<IsolateRef>>();
+    _onDartVmEvent.listen(
+      (DartVmEvent event) async {
```

Comment:
You should probably add `onDone` and `onError` params here too, to keep from waiting the entire timeout period when we know we don't need to.

Reply:
So I wasn't sure what to do about this, because right now at least, the listen method is implemented as a `Future<Null>` (see line 137) rather than a stream subscription.  At the moment the stream is returned as a broadcast, which has me a bit confused about how the implementation would work for the stream controller.

Reply:
Ok, so in this case, you're never sending error events... but you are closing the connection (at line 143), so adding an `onDone` callback here would at least allow you to handle the case of the stream closing before you've found a match.  ... and in the case of `onDone`, you don't need to worry about cancelling the subscription anyway since the stream is done.

Reply:
Hmmm, I think the `onDone` case wouldn't lead to anything but a warning log.  The only way for that event to come up would be if the user called `stop` before finding the isolate in question (which I think can only happen if the user doesn't use `await` for `getMainIsolatesByPattern` and then calls `stop`).